gto-support-lifecycle/src/main/java/org/ccci/gto/android/common/lifecycle/CollectionLiveData.kt

```diff
@@ -0,0 +1,38 @@
+package org.ccci.gto.android.common.lifecycle
+
+import androidx.lifecycle.LiveData
+
+sealed class CollectionLiveData<T, C : Collection<T>, MC : MutableCollection<T>>(
+    private val collection: MC
+) : LiveData<C>(collection as C) {
+    fun add(item: T) = collection.add(item).also { if (it) notifyChanged() }
+    fun addAll(items: Collection<T>) = collection.addAll(items).also { if (it) notifyChanged() }
+    fun remove(item: T) = collection.remove(item).also { if (it) notifyChanged() }
+    fun removeAll(items: Collection<T>) = collection.removeAll(items).also { if (it) notifyChanged() }
+
```

Comment:
Shouldn't these functions be private since the `plusAssign` etc... are present?

Reply:
no, I wanted to expose them as part of the public API. the `plusAssign` functions are operator functions which provide some syntactic sugar. In actual usage you should only really use operator functions as operators and not as functions.

so, use this: `obj += i` instead of `obj.plusAssign(i)`